### PR TITLE
Add job site and sandstone block/item convention tags

### PR DESCRIPTION
**Pull request by aws404**
Adds the following convention tags for both items/blocks:
* `c:villager_job_sites`
* `c:yellow_sandstone_blocks`
* `c:red_sandstone_blocks`
* `c:sandstone_blocks`
* `c:yellow_sandstone_slabs`
* `c:red_sandstone_slabs`
* `c:sandstone_slabs`
* `c:yellow_sandstone_stairs`
* `c:red_sandstone_stairs`
* `c:sandstone_stairs`

### CODE REVIEW

File: fabric-convention-tags-v1/src/main/java/net/fabricmc/fabric/api/tag/convention/v1/ConventionalBlockTags.java [hunk 1]
```diff
@@ -33,7 +33,21 @@ public class ConventionalBlockTags {
...
+	public static final TagKey<Block> RED_SANDSTONE_BLOCKS = register("red_sandstone_blocks");
+	public static final TagKey<Block> SANDSTONE_BLOCKS = register("sandstone_blocks");
+	public static final TagKey<Block> YELLOW_SANDSTONE_SLABS = register("yellow_sandstone_slabs");
+	public static final TagKey<Block> RED_SANDSTONE_SLABS = register("red_sandstone_slabs");
+	public static final TagKey<Block> SANDSTONE_SLABS = register("sandstone_slabs");
+	public static final TagKey<Block> YELLOW_SANDSTONE_STAIRS = register("yellow_sandstone_stairs");
+	public static final TagKey<Block> RED_SANDSTONE_STAIRS = register("red_sandstone_stairs");
+	public static final TagKey<Block> SANDSTONE_STAIRS = register("sandstone_stairs");
+
+	public static final TagKey<Block> JOB_SITE_BLOCKS = register("job_site_blocks");
```
Comment (Technici4n) [inline fabric-convention-tags-v1/src/main/java/net/fabricmc/fabric/api/tag/convention/v1/ConventionalBlockTags.java]: Isn't there already `#minecraft:acquirable_job_site` for this?

Comment (aws404) [reply fabric-convention-tags-v1/src/main/java/net/fabricmc/fabric/api/tag/convention/v1/ConventionalBlockTags.java]: `#minecraft:acquirable_job_site` is a POI tag, not a block tag. You could theoretically use it to get the blocks, however it's not very efficent

Comment (Technici4n) [reply fabric-convention-tags-v1/src/main/java/net/fabricmc/fabric/api/tag/convention/v1/ConventionalBlockTags.java]: Ah, nevermind then. I think a better name would be `#c:villager_job_sites` to explain that it's about villagers. The `_blocks` suffix is unnecessary here imo.

File: fabric-convention-tags-v1/src/datagen/java/net/fabricmc/fabric/impl/tag/convention/datagen/generators/BlockTagGenerator.java [hunk 2]
```diff
@@ -112,4 +114,59 @@ private void generateGlassTags() {
...
+				.add(Blocks.CUT_SANDSTONE)
+				.add(Blocks.CHISELED_SANDSTONE)
+				.add(Blocks.SMOOTH_SANDSTONE);
+		getOrCreateTagBuilder(ConventionalBlockTags.RED_SANDSTONE_BLOCKS)
+				.add(Blocks.RED_SANDSTONE)
+				.add(Blocks.CUT_RED_SANDSTONE)
+				.add(Blocks.CHISELED_RED_SANDSTONE)
+				.add(Blocks.SMOOTH_RED_SANDSTONE);
+		getOrCreateTagBuilder(ConventionalBlockTags.SANDSTONE_BLOCKS)
+				.addOptionalTag(ConventionalBlockTags.YELLOW_SANDSTONE_BLOCKS)
```
Comment (Technici4n) [inline fabric-convention-tags-v1/src/datagen/java/net/fabricmc/fabric/impl/tag/convention/datagen/generators/BlockTagGenerator.java]: Why optional btw, just curious?

**Review by Technici4n — APPROVED** [review]

**Comment by haykam821** [comment]
I don't think regular sandstone should be referred to as yellow sandstone. Is there a use case for having a combined sandstone tag? Since sandstone and red sandstone are clearly different blocks that wouldn't make sense to be part of the same tag, I think that `c:yellow_sandstone_blocks` and similar tags should be renamed to `c:sandstone_blocks`.

**Comment by Technici4n** [comment]
It is quite common, on Forge at least, to be able to use red sand in recipes that use sand. So would it make sense to be able to use both kinds of sandstone with a common tag? Not sure.

**Comment by aws404** [comment]
> I don't think regular sandstone should be referred to as yellow sandstone. Is there a use case for having a combined sandstone tag? Since sandstone and red sandstone are clearly different blocks that wouldn't make sense to be part of the same tag, I think that `c:yellow_sandstone_blocks` and similar tags should be renamed to `c:sandstone_blocks`.

> It is quite common, on Forge at least, to be able to use red sand in recipes that use sand. So would it make sense to be able to use both kinds of sandstone with a common tag? Not sure.

This was along my trail of thought, happy to change it if the consensus is otherwise though.

File: fabric-convention-tags-v1/src/datagen/java/net/fabricmc/fabric/impl/tag/convention/datagen/generators/BlockTagGenerator.java (hunk 2 above)
Comment (aws404) [reply fabric-convention-tags-v1/src/datagen/java/net/fabricmc/fabric/impl/tag/convention/datagen/generators/BlockTagGenerator.java]: I'm not 100% sure, I used optional because thats what everything else uses. I can't see a reason, but I assumed there was one.

**Comment by dexman545** [comment]
As tags are a collection, there is ambiguity if `sandstone` means all sandstones or just the original sandstone. Unless the collection of all sandstones is given some demarcating feature, the original sandstone collections should be marked as `yellow`,  `normal`, or other.

**Comment by haykam821** [comment]
Sandstone isn't referred to as yellow within vanilla Minecraft, so it shouldn't be referred to as yellow within tags.

I think that any tags containing only sandstone and not red sandstone should be prefixed with 'normal' or similar, or the sandstone and red sandstone combined tags should be prefixed with 'all' or similar.

File: fabric-convention-tags-v1/src/datagen/java/net/fabricmc/fabric/impl/tag/convention/datagen/generators/BlockTagGenerator.java [hunk 3]
```diff
@@ -112,4 +114,59 @@ private void generateGlassTags() {
...
+				.add(Blocks.FLETCHING_TABLE)
+				.add(Blocks.GRINDSTONE)
+				.add(Blocks.LECTERN)
+				.add(Blocks.LOOM)
+				.add(Blocks.SMITHING_TABLE)
+				.add(Blocks.SMOKER)
+				.add(Blocks.STONECUTTER);
+	}
+
+	private void generateSandstoneBlocksTag() {
```
Comment (Juuxel) [inline fabric-convention-tags-v1/src/datagen/java/net/fabricmc/fabric/impl/tag/convention/datagen/generators/BlockTagGenerator.java]: Nitpick: the name should be plural (`...Tags`)

**Review by Juuxel — APPROVED** [review]

**Comment by Player3324** [comment]
I'm not a huge fan of the normal prefix - it's only the color being "normal", but not the structure/naturalness. Uncolored or so might be better. The slabs and stairs lack the prefix while the blocks have it.

File: fabric-convention-tags-v1/src/main/java/net/fabricmc/fabric/api/tag/convention/v1/ConventionalItemTags.java [hunk 4]
```diff
@@ -16,6 +16,7 @@
 
 package net.fabricmc.fabric.api.tag.convention.v1;
 
+import net.minecraft.block.Block;
```
Comment (Player3324) [inline fabric-convention-tags-v1/src/main/java/net/fabricmc/fabric/api/tag/convention/v1/ConventionalItemTags.java]: This import is unused, checkstyle hates it ;)

**Comment by Juuxel** [comment]
> The slabs and stairs lack the prefix while the blocks have it.

There are both "normal"/uncoloured/yellow and "all" (unprefixed) versions of the slab and stair tags, though?

**Comment by Player3324** [comment]
Right, I missed that, so only the proper prefix choice question remains.

**Comment by Technici4n** [comment]
@aws404 Sorry for the delay! Could you please change the naming back to yellow? (We had some discussion on discord... Normal doesn't really work, and we couldn't find a better alternative). Should be fine after that.

**Comment by TiphaineLAURENT** [comment]
I don't see a reason to add a special these special sandstone tags. A sandstone stairs should have 3 tags : sandstone (material), stairs (type) and yellow (color) so it would work for recipes which accept sandstone or recipes that only accept yellow sandstone.

TLDR: We should use composite tag not special tag

**Comment by LambdAurora** [comment]
Someone gonna has to explain me how in hell the base sandstone is yellow.

**Comment by LambdAurora** [comment]
I want someone to look at me straight in the eyes and tell me that normal terracotta is actually brown terracotta and that the normal shulker box is actually purple shulker box. While there's already a specific dyed variant for those colors.

image

**Comment by TiphaineLAURENT** [comment]
> Someone gonna has to explain me how in hell the base sandstone is yellow.

If we have to assign it a Minecraft color (https://minecraft.fandom.com/wiki/Dye), it's definitely yellow

**Comment by LambdAurora** [comment]
> If we have to assign it a Minecraft color (https://minecraft.fandom.com/wiki/Dye), it's definitely yellow

It's called sand, not yellow_sand.

**Comment by Technici4n** [comment]
Another update, this time final I hope, after some more discussion on Discord: please change to `uncolored` for the regular sandstone.

File: fabric-convention-tags-v1/src/datagen/java/net/fabricmc/fabric/impl/tag/convention/datagen/generators/BlockTagGenerator.java [hunk 5]
```diff
@@ -112,4 +114,59 @@ private void generateGlassTags() {
...
+				.addOptionalTag(ConventionalBlockTags.RED_SANDSTONE_SLABS);
+
+		getOrCreateTagBuilder(ConventionalBlockTags.NORMAL_SANDSTONE_STAIRS)
+				.add(Blocks.SANDSTONE_STAIRS)
+				.add(Blocks.SMOOTH_SANDSTONE_STAIRS);
+		getOrCreateTagBuilder(ConventionalBlockTags.RED_SANDSTONE_STAIRS)
+				.add(Blocks.RED_SANDSTONE_STAIRS)
+				.add(Blocks.SMOOTH_RED_SANDSTONE_STAIRS);
+		getOrCreateTagBuilder(ConventionalBlockTags.SANDSTONE_STAIRS)
+				.addOptionalTag(ConventionalBlockTags.NORMAL_SANDSTONE_SLABS)
```
Comment (ghost) [inline fabric-convention-tags-v1/src/datagen/java/net/fabricmc/fabric/impl/tag/convention/datagen/generators/BlockTagGenerator.java:169]: I think this must be stairs, not slabs... 🤔

File: fabric-convention-tags-v1/src/datagen/java/net/fabricmc/fabric/impl/tag/convention/datagen/generators/ItemTagGenerator.java [hunk 6]
```diff
@@ -312,6 +314,61 @@ private void generateToolTags() {
...
+				.addOptionalTag(ConventionalItemTags.RED_SANDSTONE_SLABS);
+
+		getOrCreateTagBuilder(ConventionalItemTags.NORMAL_SANDSTONE_STAIRS)
+				.add(Items.SANDSTONE_STAIRS)
+				.add(Items.SMOOTH_SANDSTONE_STAIRS);
+		getOrCreateTagBuilder(ConventionalItemTags.RED_SANDSTONE_STAIRS)
+				.add(Items.RED_SANDSTONE_STAIRS)
+				.add(Items.SMOOTH_RED_SANDSTONE_STAIRS);
+		getOrCreateTagBuilder(ConventionalItemTags.SANDSTONE_STAIRS)
+				.addOptionalTag(ConventionalItemTags.NORMAL_SANDSTONE_SLABS)
```
Comment (ghost) [inline fabric-convention-tags-v1/src/datagen/java/net/fabricmc/fabric/impl/tag/convention/datagen/generators/ItemTagGenerator.java:368]: I think this must be stairs, not slabs... 🤔

**Review by ghost — CHANGES_REQUESTED** [review]
My feedback is that i agree to change the name to "uncolored" instead of normal, because "normal"  it too much generic and means anything (normal =  no danger; normal = no shape; normal = no food.... etc. this is an example 😅), "uncolored" refers only to colors and is the correct name.

**Comment by Technici4n** [comment]
Replaced by #3051.
